yarn-project/p2p/src/services/peer-manager/peer_manager.ts

```diff
@@ -396,6 +455,20 @@ export class PeerManager implements PeerManagerInterface {
       this.authenticatedPeerIdToValidatorAddress.has(peerIdAsString)
     );
   }
+
+  /*
+   * Checks whether peer is allowed to connect
+   *
+   * @param addresses: Address of the node
+   *
+   * @returns: True if node is allowed to connect, otherwise false
+   * */
+  public isNodeAllowedToConnect(id: string | PeerId): boolean {
+    const failedAuthHandshakesCount = this.failedAuthHandshakes.get(id.toString()) ?? 0;
```

Comment:
I'm thinking we should remove failures after a period of time. We could have situations where a validator is removed from the set, gets banned, is added back to the set and then can't ever connect to the preferred peers again.